### ISSUE TITLE
Add support for Java 11

### ISSUE THREAD

**Issue by beatngu13** [retest/recheck#150]
While source compatability should be Java 8+, we should also add support running on Java 11 (the current LTS). Running `mvn verify` in recheck-web yields:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.0.1:jar (bundle-javadoc) on project recheck-web: MavenReportException: Error while generating Javadoc: 
[ERROR] Exit code: 1 - warning: unknown enum constant XmlAccessType.FIELD
[ERROR]   reason: class file for javax.xml.bind.annotation.XmlAccessType not found
```

This is because JAXB has been removed as of Java 9 (I will soon provide a fix for that).

However, as already suggested by @modulo11, I think we should compile with Java 8 and run the tests with Java 11 for proper feedback.

Other suggestions?

**Comment by beatngu13** [comment]
#151 appears to have solved all compilation and runtime issues with Java 11. As soon as we have adapted our Travis CI pipeline to compile with 8 and run tests on 11, this can be closed.

**Comment by martin-v** [comment]
Is implemented, but should be integrated in build pipeline.

**Comment by beatngu13** [comment]
After Varun's fix for the `javadoc` tool (#217), everything seems to be fine. I've downloaded [today's beta release](https://search.maven.org/artifact/de.retest/recheck/1.1.0-beta.1/jar) and the `MANIFEST.MF` says:

```
Manifest-Version: 1.0
Archiver-Version: Plexus Archiver
Built-By: travis
Created-By: Apache Maven 3.5.2
Build-Jdk: 1.8.0_151
```

That is, Travis appears to automatically switch back to Java 8 for deployment. So I would say we can apply the same concept to recheck-web and then close this ticket.
